enums.rb

```diff
@@ -0,0 +1,94 @@
+# rubocop:disable Style/NestedTernaryOperator, Style/MultilineTernaryOperator, Style/ParallelAssignment, Style/IfUnlessModifier, Style/ExplicitBlockArgument, Style/For, Metrics/CyclomaticComplexity
+module Enumerable
+  def my_each
+    return to_enum(:my_each) unless block_given?
+
+    for i in self
+      yield i
+    end
+  end
+
+  def my_each_with_index
+    return to_enum(:my_each_with_index) unless block_given?
+
+    index = 0
+    for i in self
+      yield i, index
+      index += 1
+    end
+  end
+
+  def my_select
+    arr = []
+    return to_enum(:my_select) unless block_given?
+
+    my_each { |n| arr.push(n) if yield n }
+    arr
+  end
+
+  def my_all?(*args)
+    case args.size
+    when 0
+      block_given? ? my_select { |n| yield n }.size == size : false
+    when 1
+      args.first.is_a?(Regexp) ?
+      my_select { |n| n.match(args.first) }.size == size :
+      my_select { |n| n.is_a?(args.first) }.size == size
+    end
+  end
+
+  def my_any?(*args)
+    case args.size
+    when 0
+      block_given? ? my_select { |n| yield n }.size.positive? : false
+    when 1
+      args.first.is_a?(Regexp) ?
+      my_select { |n| n.match(args.first) }.size.positive? :
+      my_select { |n| n.is_a?(args.first) }.size.positive?
+    end
+  end
+
+  def my_none?(*args)
+    case args.size
+    when 0
+      block_given? ? !my_select { |n| yield n }.size.positive? : !my_select { |n| n == true }.size.positive?
+    when 1
+      args.first.is_a?(Regexp) ?
+      !my_select { |n| n.match(args.first) }.size.positive? :
+      !my_select { |n| n.is_a?(args.first) }.size.positive?
+    end
+  end
+
+  def my_count(*args)
+    case args.size
+    when 0
+      block_given? ? my_select { |n| yield n }.size : size
+    when 1
+      block_given? ? my_select { |n| yield(args.first, n) && n == args.first }.size : my_select do |n|
+                                                                                        n == args.first
+                                                                                      end.size
+    end
+  end
+
+  def my_map(proc = nil)
+    arr = []
+    my_each { |n| arr.push(yield(n)) } if block_given?
+    my_each { |n| arr.push(proc.call(n)) } if proc
+    arr
+  end
```

Comment:
- #my_map returns an Enumerator if no block is given. It throws this error: `expected [] to be a kind of Enumerator`. I expected  a similar code like this: return `enum_for unless block_given?` at the top of `arr = []`. This would have taken care of this error. Pls fix this.
- #my_map executes only the proc when both a block and a proc are given and returns a new array with the results of running the proc once for every element in enum. One the results is returning `false` instead of `true`